src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java

```diff
@@ -519,22 +522,35 @@ public List<BitbucketCloudBranch> getBranchesByRef(String nodePath) throws IOExc
     @Override
     @CheckForNull
     public BitbucketCommit resolveCommit(@NonNull String hash) throws IOException, InterruptedException {
-        String url = UriTemplate.fromTemplate(REPO_URL_TEMPLATE + "/commit/{hash}")
-                .set("owner", owner)
-                .set("repo", repositoryName)
-                .set("hash", hash)
-                .expand();
-        String response;
+        final String url = UriTemplate.fromTemplate(REPO_URL_TEMPLATE + "/commit/{hash}")
+            .set("owner", owner)
+            .set("repo", repositoryName)
+            .set("hash", hash)
+            .expand();
+
+        Callable<BitbucketCloudCommit> request = () -> {
+            String response;
+            try {
+                response = getRequest(url);
+            } catch (FileNotFoundException e) {
+                return null;
+            }
+            try {
+                return JsonParser.toJava(response, BitbucketCloudCommit.class);
+            } catch (IOException e) {
+                throw new IOException("I/O error when parsing response from URL: " + url, e);
+            }
+        };
+
         try {
-            response = getRequest(url);
-        } catch (FileNotFoundException e) {
+            if (enableCache) {
+                return cachedCommits.get(hash, request);
+            } else {
+                return request.call();
+            }
+        } catch (Exception ex) {
```

Comment:
If I read right this catch eat any exception thrown by the callable. Before exception was log to the build console, now it is not logged and return null

Reply:
Hm, I think you might be right. Did you already create a new issue for this, as this PR is already merged?

Reply:
no, I was here because I check changes in jenkins plugin before update in production.
Feel free to create an issue and a PR with a unit test

Reply:
@thomas-boehm-tractive would you be able to create a PR?

Reply:
@thomas-boehm-tractive any news about fix for this issue?

Reply:
Sorry for the late response, I didn't get to this in the last weeks.
Honestly, I am not really sure if this really is an issue, as similar blocks in this file behave exactly like this (see e.g. getTeam or getTeamAvatar function). My PR changed the behavior, but change it to behave like the code does on several other positions.
Are you convinced I should change this?

Reply:
Yes I'm sure, what is the reason to eat the exception and not inform the user that something was wrong?
I was on these changes of this pull request, I did see other places. Eventually we should fix also other code.
In short what should do the jenkins user if something happens and there is no apparently reason? I should replicate with jenkins in debug mode

Reply:
Created a ticket, and will create a PR within the next days: 
https://github.com/jenkinsci/bitbucket-branch-source-plugin/issues/850

Reply:
Created PR, I think this should bring back the behavior from before this PR was merged: https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/851